test/end-to-end/vehicle.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const nightwatch = require('nightwatch');
+const test = require('ava');
+
+const smartcar = require('../../');
+
+const {getAuthClientParams, runAuthFlow} = require('./helpers');
+const config = require('../config');
+
+const context = {};
+
+test.before(async() => {
+  context.client = nightwatch.initClient(config.get('nightwatch'));
+  context.browser = context.client.api();
+
+  const client = new smartcar.AuthClient(getAuthClientParams());
+
+  const authUrl = client.getAuthUrl();
+
+  const code = await runAuthFlow(context.client, context.browser, authUrl);
+
+  const access = await client.exchangeCode(code);
+  const accessToken = access.accessToken;
+  const vehicleIds = await smartcar.getVehicleIds(accessToken);
+  context.vehicle = new smartcar.Vehicle(vehicleIds.vehicles[0], accessToken);
+});
+
+test('vehicle info', async(t) => {
+  t.notThrows(async() => await context.vehicle.info());
+});
+
+test('vehicle location', async(t) => {
+  t.notThrows(async() => await context.vehicle.location());
+});
+
+test('vehicle odometer', async(t) => {
+  t.notThrows(async() => await context.vehicle.odometer());
+});
+
+test('vehicle vin', async(t) => {
+  t.notThrows(async() => await context.vehicle.vin());
+});
+
+test('vehicle lock', async(t) => {
+  t.notThrows(async() => await context.vehicle.lock());
+});
+
+test('vehicle unlock', async(t) => {
+  t.notThrows(async() => await context.vehicle.unlock());
+});
+
+test('vehicle permissions', async(t) => {
+  t.notThrows(async() => await context.vehicle.permissions());
+});
+
+test('vehicle disconnect', async(t) => {
+  t.notThrows(async() => await context.vehicle.disconnect());
```

Comment:
I don't think these tests work correctly as written reason being that the `t.notThrows` is not being `await`'d 

The vehicles methods return promises and we await that in the callback function and then `t.notThrows` awaits the callback function, but nothing waits for the promise that `t.notThrows` returns.

Since `t.notThrows` accepts a promise, the callback isn't really needed either, as such you should do one of the following:

```js
test('name', async(t) => {
  await t.notThrows(context.vehicle.method());
});
```

or

```js
test('name', (t) => {
  return t.notThrows(context.vehicle.method());
});
```

There are two tests in index.js which do something similar please fix those as well.